source/common/res/features/budget-category-info/main.js

```diff
@@ -160,7 +160,8 @@
           // when this is moved to the webpack codebase, hopefully we can just listen to onRouteChanged
           // but until then we'll just have to keep adding stuff to this check that should trigger
           // budget-category related features to update
-          if (changedNodes.has('navlink-budget active') ||
+          if ((changedNodes.has('nav-main') && ynabToolKit.shared.getCurrentRoute().includes('budget')) ||
```

Comment:
nav-main will be changed because it has children removed and created, and when nav-main changes, we want to make sure our route is 'budget', so as to not run our logic when not on the budget page.